### PR TITLE
Improve OS detection and SSH configuration in Packet

### PR DESCRIPTION
**Pull request by nacx**
* Adds Alpine Linux as a known OSFamily to jclouds-compute, and improves the OS detection in Packet.
* Packet has an API to set the key pairs, so this PR removes the default script that implicitly uploads the key via SSH if it is configured in the options.
* Also removes a couple unused classes.

### CODE REVIEW

File: providers/packet/src/main/java/org/jclouds/packet/domain/ActionType.java [hunk 1]
```diff
@@ -1,39 +0,0 @@
...
-
-/**
- * Performs an action for the given device. Possible actions include:
-
- power_on
- power_off
- reboot
- rescue: reboot the device into rescue OS.
- */
-public enum ActionType {
```
Comment (andreaturli) [inline providers/packet/src/main/java/org/jclouds/packet/domain/ActionType.java]: is this not needed anymore?

File: providers/packet/src/main/java/org/jclouds/packet/compute/config/PacketComputeServiceContextModule.java [hunk 2]
```diff
@@ -80,6 +81,7 @@ protected void configure() {
       });
       bind(TemplateOptions.class).to(PacketTemplateOptions.class);
       bind(CreateNodesInGroupThenAddToSet.class).to(CreateSshKeysThenCreateNodes.class);
+      bind(NodeAndTemplateOptionsToStatement.class).to(NodeAndTemplateOptionsToStatementWithoutPublicKey.class);
```
Comment (andreaturli) [inline providers/packet/src/main/java/org/jclouds/packet/compute/config/PacketComputeServiceContextModule.java]: didn't know that! :)

looks quite useful! thanks

File: providers/packet/src/main/java/org/jclouds/packet/domain/ActionType.java (hunk 1 above)
Comment (nacx) [inline providers/packet/src/main/java/org/jclouds/packet/domain/ActionType.java]: It wasn't used anywhere

Comment (andreaturli) [inline providers/packet/src/main/java/org/jclouds/packet/domain/ActionType.java]: ok :D

**Comment by andreaturli** [comment]
lgtm +1

**Comment by nacx** [comment]
Merged to master and 2.0.x. The packet specific parts for 2.0.x have been pushed to the jclouds-labs repo.

**Pull request closed without merging**